### DISCUSSION TITLE
Border Router - Thread credentials management by Android Apps

### DISCUSSION THREAD

**Discussion by doru91** [openthread/openthread#9243] [Q&A] [answered]
This question is in context of [Thread Network SDK](https://developers.home.google.com/thread) for Android:

- currently third part apps can connect to the Border Agent and set Thread credentials/get Thread credentials/create Thread network using preferred credentials;
- discovery of the BA is possible through MeshCop mDNS advertisments if [OTBR_PUBLISH_MESHCOP_BA_ID ](https://github.com/openthread/ot-br-posix/pull/1832/files#diff-2dfd8d58f7ec7b21e573c09220dc27e20c77fb6fea1afd92e370c90b1a8f4f8a) is defined in otbr posix code. This will further enable "id" entry in the mDNS advertismments. However, in Thread 1.3 spec this is not defined under "Table 8-4. DNS-SD TXT Answer Record Key-Value Pair Format". Is this "id" entry out of spec and defined by Google?
- once the BA is discovered how is the connection established between Android App (Mobile Phone) and BA? Which port/protocols are used?

At a first glance, all this mechanism seems to be proprietary and I would like to understand if this is the case and on whether we can formalize it in Matter/Network Commissioning Cluster context.

Thanks a lot!

**Comment by jwhui** [comment, marked as answer]
> However, in Thread 1.3 spec this is not defined under "Table 8-4. DNS-SD TXT Answer Record Key-Value Pair Format". Is this "id" entry out of spec and defined by Google?

Thread 1.3.0 reserves the `id` TXT Key and a spec update will fully specify it. See [SPEC-1020](https://threadgroup.atlassian.net/browse/SPEC-1020) and [SPEC-1191](https://threadgroup.atlassian.net/browse/SPEC-1191).

> once the BA is discovered how is the connection established between Android App (Mobile Phone) and BA? Which port/protocols are used?

The Thread Network SDK for Android does not generally connect to Thread Border Routers. It is simply a way to share Thread network credentials between apps on the Android device. How Thread Credentials are retrieved from Thread Border Routers is currently vendor specific.

> At a first glance, all this mechanism seems to be proprietary and I would like to understand if this is the case and on whether we can formalize it in Matter/Network Commissioning Cluster context.

Matter could be one option to standardize communication of Thread Network Credentials. Of course, that would be specific to devices that implement Matter.
